config/config.go

```diff
@@ -108,11 +109,17 @@ func NewFromReader(reader io.Reader) (S3Cli, error) {
 		return S3Cli{}, fmt.Errorf("Invalid credentials_source: %s", c.CredentialsSource)
 	}
 
+	// For Alibaba Cloud, host is required.
 	if c.Region == "" && c.Host == "" {
 		c.Region = defaultRegion
 	}
-	if c.Region == "" && c.Host != "" && c.isAWSHost() {
-		c.Region = c.getRegionFromHost()
+	if c.Region == "" && c.Host != "" {
+		if c.isAWSHost() {
+			c.Region = c.getRegionFromHost()
+		}
+		if c.isAlicloudHost() {
+			c.Region = c.getAlicloudRegionFromHost()
```

Comment:
Out of curiosity, do we need to set the region on the AWS SDK to make this work with Alicloud? Does it work without it? If we don't need it it would simplify a lot of this PR.

Reply:
I think, do some work in the aws sdk is not good solution. I prefer to add a provider field in the config, and region can be distinguished via the provider. Also, users can change different blobstore based on provider field in the manifest, like following:

```
    provider: oss
    bucket_name: bosh-alicloud-cpi-blobs
    region: cn-hangzhou
    access_key_id: ((access_key_id))
    secret_access_key: ((access_key_secret))
```